lib/board.rb

```diff
@@ -1,33 +1,63 @@
 class Board 
-	attr_accessor :matrix
-	def initialize
-		@matrix = [nil, nil, nil, nil, nil, nil, nil, nil, nil]
+	attr_accessor :matrix, :win_combinations, :dim
+	def initialize(dim)
+		@matrix = Array.new(dim**2)
+		@win_combinations = init_combin(dim)
+		@dim = dim
+	end
+  
+  	def init_combin(dim)
+  		arr = []
+        arr  += Array.new(dim ){|l| Array.new(dim){|i| dim * l + i }}  
+        arr  += Array.new(dim){|l| Array.new(dim){|i|  l + (i*dim)}}
+        arr  += Array.new(1){|l| Array.new(dim){|i|  i+ (i*dim)}}
+        arr  += Array.new(1){|l| Array.new(dim){|i|  i*(dim-1) + (dim-1)}}
+        return arr                      
+    end
+
+    def valid_input?(arr)
+		if arr.length == 2 && 
+		(ALPHABET[0...@dim].any? {|val| val == arr[1].upcase}) &&
+		((0...@dim).to_a.any? {|element| element == arr[0].to_i})
+			return true
+		end
+		false
+	end
+
+	def empty_space?(i)
+		return true if @matrix[i] == nil
+		false
+	end
+
+    def full?
+		@matrix.none? {|val| val == nil} 
+	end
+
+	def player_won?(symbol)
+		@win_combinations.each do |element|
+	       return true if element.all?{|val| @matrix[val] == symbol}                    
+      	end
+		false
 	end
 
 	def print_board
 		counter = 0
-		@matrix.each do |val|
-			print "|" if counter == 0
-			
-			if val
-				print " #{val} "
-			else
-				print "   "
-			end
+		print "  "
+		@dim.times {|i| print " #{ALPHABET[i]} " }
+		print "\n"
+		@matrix.each_slice(@dim) do |val|
+			print "#{counter}|"
+			val.each {|element| print element == nil ? " - " : " #{element} "} 
+			print "|\n"
 			counter+=1
-			if counter==3
-				counter = 0
-				print "|\n" 	
-			end
-		end 
+		end
 	end
 
-	def change_matrix(symbol, index)
-		@matrix[index] = symbol
+	def change_matrix(symbol, move_index)
+		@matrix[move_index] = symbol
 	end
 
 	def clear
-		@matrix.each {|element| element = nil}
+		@matrix = Array.new(dim**2)
```

Comment:
You forgot the `@` before `dim`. Also, do you really need to call `Array.new(number)`? I mean, is it necessary that you have an array full of `nil` values?